backend/src/routes/users.js

```diff
@@ -117,26 +117,26 @@ router.post('/:userId/block', async (req, res, next) => {
     );
     
     // Update list for second user and increment blocked count
-    matchUser = await User.updateOne(
+    blockedUser = await User.updateOne(
         { _id: req.body.blockedId },
         { $push: { notRecommended: req.params.userId }, $inc: { blockedCount: 1 } },
         { new: true },
     );
     
     // Delete and cascade for the user if blocked count meets threshold
-    if (matchUser && matchUser.blockedCount >= BLOCK_THRESHOLD){
+    if (blockedUser && blockedUser.blockedCount >= BLOCK_THRESHOLD){
         // Error handling should not happen here considering that we already know the user exists
-        const deleteUserId = matchUser.userId;
+        const deleteUserId = blockedUser.userId;
         await User.findByIdAndDelete(deleteUserId);
         await Listing.deleteMany({ deleteUserId });
         await Preferences.deleteOne({ deleteUserId });
```

Comment:
Not sure if these functions have return values or not, but if so, then we would probably be able to rollback transactions if these are null/false/whatever. If not then, that's okay.

Reply:
They are all wrapped inside a transaction so I dont think this should be a problem no?